src/main/scala-2.11/markets/participants/LiquidityDemander.scala

```diff
@@ -15,53 +15,43 @@ limitations under the License.
 */
 package markets.participants
 
-import akka.actor.Scheduler
 
-import markets.orders.Order
-
-import scala.concurrent.ExecutionContext
-import scala.concurrent.duration.FiniteDuration
+import markets.orders.market.{MarketAskOrder, MarketBidOrder}
+import markets.tradables.Tradable
 
 
 /** A Trait providing behavior necessary to submit `MarketOrderLike` orders. */
 trait LiquidityDemander extends MarketParticipant {
 
-  def generateMarketOrder(): Order
+  def marketAskOrderStrategy(): Option[(Long, Tradable)]
+
+  def marketBidOrderStrategy(): Option[(Long, Tradable)]
 
-  /** Schedule a market order.
-    *
-    * @param scheduler
-    * @param initialDelay
-    * @param executionContext
-    */
-  def scheduleMarketOrder(scheduler: Scheduler,
-                          initialDelay: FiniteDuration)
-                         (implicit executionContext: ExecutionContext): Unit = {
-    scheduler.scheduleOnce(initialDelay, self, SubmitMarketOrder)(executionContext)
+  private final def generateMarketAskOrder(quantity: Long, tradable: Tradable) = {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [generateMarketAskOrder should have a smaller scope by using private[this]](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/1703602213/issues/source?bid=2921361&fileBranchId=3046267#l30)